src/librustc_errors/lib.rs

```diff
@@ -42,6 +42,8 @@ use std::cell::{RefCell, Cell};
 use std::mem;
 use std::rc::Rc;
 use std::{error, fmt};
+use std::sync::atomic::AtomicUsize;
```

Comment:
`err_count` probably needs to be done in a smarter way than here to assure thread-safety, so could you add a `FIXME`? In any case this is better than the alternative.

Reply:
In which case is this not thread safe?